com.unity.multiplayer.mlapi/Tests/Editor/InterestTests.cs

```diff
@@ -0,0 +1,344 @@
+using System.Collections.Generic;
+using UnityEngine;
+using NUnit.Framework;
+using MLAPI.Connection;
+using MLAPI.Configuration;
+using MLAPI.Interest;
+
+namespace MLAPI.EditorTests
+{
+    public class TestInterestSettings : InterestSettings
+    {
+        public int SomeSetting;
+    }
+
+    public class InterestTests
+    {
+        public class DelegateTester : InterestNode
+        {
+            public DelegateTester()
+            {
+                OnSpawn = delegate
+                {
+                   SpawnCalled++;
+                };
+                OnDespawn = delegate
+                {
+                   DespawnCalled++;
+                };
+            }
+
+            public int SpawnCalled = 0;
+            public int DespawnCalled = 0;
+        }
+
+        public class OddsEvensStorage : InterestObjectStorage
```

Comment:
As a user that wants to create their own logic for these, I would need to know about both storage and nodes right? I’m curious why do we have both the concepts for nodes and containers? Why not just node? and inherit from node to create our custom behaviour (no need for container) instead of having to create a custom container and then add it to a new instance of a node? that’d make one less class users would need to worry about?